.github/workflows/e2e-tests-main-devnet.yml

```diff
@@ -28,7 +28,7 @@ jobs:
   build-test-docker:
     needs: [build-new-node]
     name: Build docker image with the test node artifact
-    runs-on: ubuntu-20.04
+    runs-on: self-hosted
```

Comment:
can we please rename this confusing workflow?

Reply:
Could you please provide an alternative name? It looks good to me now

Reply:
I can't comment on the whole file, only on changed lines - I mean the whole workflow file name `e2e-tests-main-devnet` is confusing as hell. Is it mainnet? Or is it devnet?

Reply:
I'm afraid this is not the scope of this PR, as we want to focus on changes that decrease workflow time here.

Good point nevertheless, we have plans to rework workflows anyway, so I promise we'll take that into account.

Reply:
I've added https://cardinal-cryptography.atlassian.net/browse/A0-2043